src/formatDate.js

```diff
@@ -8,7 +8,46 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const arrayFromDate = date.split(fromFormat[fromFormat.length - 1]);
+  const arrForResulf = [];
+  const objectFromDate = {};
+
+  for (let i = 0; i < arrayFromDate.length; i++) {
+    objectFromDate[fromFormat[i]] = arrayFromDate[i];
+  }
+
+  const { MM: month, DD: day, ...lastElement } = objectFromDate;
+  const [year] = Object.values(lastElement);
+
+  for (const y of toFormat) {
+    if (y === 'YYYY' || y === 'YY') {
+      if (year.length === y.length) {
```

Comment:
The condition `year.length === y.length` assumes that the year in the input date will always match the length of the format string. This may not be true if the input date uses a different year format than expected. Ensure that the input date is validated against the `fromFormat` before processing.